### ISSUE TITLE
Domain request: .gov `domain step form field` needs visual indicator that field is required

### ISSUE THREAD

**Issue by AnnaGingle** [cisagov/manage.get.gov#2357]
### Issue description

When requesting a new .gov domain, the .gov step's _What .gov domain do you want?_ input field is required but there is no visual indication. This page would fail trusted tester ID 2.4.6-label-descriptive making it non-compliant with section 508.
![Screenshot 2024-06-24 at 9 43 57 AM](https://github.com/cisagov/manage.get.gov/assets/48488699/afc5eec2-eb6b-47c8-b0a7-778f8ac3e067)

### Acceptance criteria

- [ ] The .gov step's _What .gov domain do you want?_ input field provides a visual indication that it is required

![Screenshot 2024-06-24 at 9 47 41 AM](https://github.com/cisagov/manage.get.gov/assets/48488699/77229cbe-dd5e-4d63-907b-d9e60355ed33)

**Comment by PaulKuykendall** [comment]
@Katherine-Osos is speaking with design regarding patterns

**Comment by AnnaGingle** [comment]
This is a duplicate ticket. This work is included in [Required fields in the request form are marked as required](https://github.com/cisagov/manage.get.gov/issues/2698). Closing ticket.
